contentful/apps
# feat: add auth flow for MS Teams [INTEG-1761]

## Purpose

This PR adds an authorization flow for the MS Teams app so that we can get the tenant id from an authorized MS user rather than having the user paste it in.

## Approach

I utilized the [Microsoft Authentication Library for React](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/README.md#samples). I implemented an `AuthProvider` which allows us to use hooks and methods from this library to authenticate a user. 

When a user authenticates on initial installation, I am using our `customApi.saveConfiguration` to trigger an installation and save the tenant id in installation parameters. 

The `Disconnect` button prompts a modal asking the user to confirm that they want to sign out, then goes into the MS logout flow. If a user is not authenticated, then the notifications section won't show.

For now, I am displaying the name and username of the authenticated user. We have a follow up ticket to get the organization name and logo.

## Testing steps

You will need to add `VITE_MS_APP_CLIENT_ID` environment variable (see 1password).

## Breaking Changes

## Dependencies and/or References

## Deployment
